### PR TITLE
Changes to support scoring and rendering good videos on the servers

### PR DESCRIPTION
**Pull request by grymmy**
This PR does a few things, most specifically it supports generating reasonable-looking videos on the current DO server.  Here are the following changes:

- Changes the resolution/video encoding options
- Attempts to add an in-memory browser cache
- Configures puppeteer to ignore audio
- Adds more time buffer before and after starting recording to avoid undesirable clips
- Changes the method by which we detect level completion

This is an [example video](https://res.cloudinary.com/dgf5wjpcx/video/upload/v1680137364/f5gwfhqqlkh5uv5wvkjm.webm) it created.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
+// On top of your code
+let cache: { [url: string]: CacheEntry } = {};
+
 export const playLevel = async (rawLevelUrl: string, videoName: string, folder: string) => {
   const startTime = Date.now()  
-  const tickRate = 1000
-  const drawModulo = 1
+  const tickRate = 30 * 5
+  const drawModulo = 3
   const defaultTickRate = 30
```
Comment (JosiasAurel) [inline src/main.ts:17]: could it be that a very high tickRate makes puppeteer wonky?

File: src/main.ts [hunk 2]
```diff
@@ -96,7 +109,7 @@ export const playLevel = async (rawLevelUrl: string, videoName: string, folder:
 
   // To avoid chopping the end of the video prematurely, we will stop the video 1 second later, adjusted
   // for our tick rate time scaling
-  const tailWaitTimeMs = 1 * (defaultTickRate / tickRate) * 1000.0
+  const tailWaitTimeMs = 2.5 * (defaultTickRate / tickRate) * 1000.0
```
Comment (JosiasAurel) [inline src/main.ts:112]: Is the tailWait really necessary since once the game is completed we should get the value of T and expression

**Review by JosiasAurel — APPROVED** [review]
looking gtm

File: src/main.ts (hunk 2 above)
Comment (grymmy) [reply src/main.ts:112]: For some reason this seems to be necessary to get a clean tail end of the video, without it it stops short.  I agree that this isn't the most elegant solution though, it's probably just sweeping an actual problem under the rug - but it seems to work well enough for now...

File: src/main.ts (hunk 1 above)
Comment (grymmy) [reply src/main.ts:17]: It DOES!  The stupid 250 tick threshold thing I [put in here](https://github.com/hackclub/sinerider/pull/303/files#diff-ba98b2d14b7d2c8892150124ac5b7743be06c04c68ecd8e72b3a6faef5dd3aadR155) is actually bugged, and if you configure it at or above that everything starts failing! :/

**Review by JosiasAurel — COMMENTED** [review]
good

**Pull request merged**